var/spack/repos/builtin/packages/xsdk/package.py

```diff
@@ -18,91 +19,128 @@ class Xsdk(BundlePackage):
+    depends_on('pflotran@xsdk-0.5.0', when='@0.5.0')
     depends_on('pflotran@xsdk-0.4.0', when='@0.4.0')
     depends_on('pflotran@xsdk-0.3.0', when='@0.3.0')
     depends_on('pflotran@xsdk-0.2.0', when='@xsdk-0.2.0')
 
     depends_on('alquimia@develop', when='@develop')
+    depends_on('alquimia@xsdk-0.5.0', when='@0.5.0')
     depends_on('alquimia@xsdk-0.4.0', when='@0.4.0')
     depends_on('alquimia@xsdk-0.3.0', when='@0.3.0')
     depends_on('alquimia@xsdk-0.2.0', when='@xsdk-0.2.0')
 
-    depends_on('sundials@4.1.0~int64+hypre', when='@develop')
+    depends_on('sundials+superlu-dist', when='@0.5.0: %gcc@6.1:')
+    depends_on('sundials@develop~int64+hypre+petsc', when='@develop')
+    depends_on('sundials@5.0.0~int64+hypre+petsc', when='@0.5.0')
     depends_on('sundials@3.2.1~int64+hypre', when='@0.4.0')
     depends_on('sundials@3.1.0~int64+hypre', when='@0.3.0')
 
-    depends_on('plasma@18.11.1:', when='@develop %gcc@6.0:')
+    depends_on('plasma@19.8.1:', when='@develop %gcc@6.0:')
+    depends_on('plasma@19.8.1:', when='@0.5.0 %gcc@6.0:')
     depends_on('plasma@18.11.1:', when='@0.4.0 %gcc@6.0:')
 
-    depends_on('magma@2.4.0', when='@develop +cuda')
+    depends_on('magma@2.5.1', when='@develop +cuda')
+    depends_on('magma@2.5.1', when='@0.5.0 +cuda')
     depends_on('magma@2.4.0', when='@0.4.0 +cuda')
     depends_on('magma@2.2.0', when='@0.3.0 +cuda')
 
-    depends_on('amrex@develop+sundials', when='@develop %intel')
-    depends_on('amrex@develop+sundials', when='@develop %gcc')
+    depends_on('amrex@develop', when='@develop %intel')
+    depends_on('amrex@develop', when='@develop %gcc')
+    depends_on('amrex@19.08', when='@0.5.0 %intel')
+    depends_on('amrex@19.08', when='@0.5.0 %gcc')
     depends_on('amrex@18.10.1', when='@0.4.0 %intel')
     depends_on('amrex@18.10.1', when='@0.4.0 %gcc')
 
     depends_on('slepc@develop', when='@develop')
+    depends_on('slepc@3.12.0', when='@0.5.0')
     depends_on('slepc@3.10.1', when='@0.4.0')
 
+    depends_on('omega-h +trilinos', when='+trilinos +omega-h')
+    depends_on('omega-h ~trilinos', when='~trilinos +omega-h')
     depends_on('omega-h@develop', when='@develop +omega-h')
+    depends_on('omega-h@9.29.0', when='@0.5.0 +omega-h')
     depends_on('omega-h@9.19.1', when='@0.4.0 +omega-h')
 
-    depends_on('strumpack@master', when='@develop')
-    depends_on('strumpack@3.1.1', when='@0.4.0')
+    depends_on('strumpack@master', when='@develop +strumpack')
+    depends_on('strumpack@3.3.0', when='@0.5.0 +strumpack')
+    depends_on('strumpack@3.1.1', when='@0.4.0 +strumpack')
 
     depends_on('pumi@develop', when='@develop')
+    depends_on('pumi@2.2.1', when='@0.5.0')
     depends_on('pumi@2.2.0', when='@0.4.0')
 
-    depends_on('tasmanian@develop+xsdkflags+blas~openmp', when='@develop')
-    depends_on('tasmanian@develop+xsdkflags+blas+cuda+magma~openmp', when='@develop +cuda')
+    tasmanian_openmp = '~openmp' if sys.platform == 'darwin' else '+openmp'
```

Comment:
Couple of comments here. First off, it would be more correct to do something like `self.spec.satisfies('%clang platform=darwin')` since other compilers have OpenMP support, but I don't think there's any way to do that here. Second of all, you _can_ actually get OpenMP working with Apple Clang if you add a dependency on `llvm-openmp` and use the right build flags. See `py-scikit-learn` and #13669 for examples of this.

Reply:
Thanks! for one - we can't access self.spec in the class constructor - hence this change.
And `lvm-openmp` option looks good. So I think we should use this.

But I get build errors with tasmanian cc: @mkstoyanov 

```
==> [2019-11-11-13:26:46.695709] 'cmake' '/var/folders/xk/v0fd809s2y7b3x8bp05l2nww0000gq/T/balay/spack-stage/spack-stage-tasmanian-7.0-u55sduzuuzqw77kbq42hjchs72qx7cpp/spack-src' '-G' 'Unix Makefiles' '-DCMAKE_INSTALL_PREFIX:PATH=/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/tasmanian-7.0-u55sduzuuzqw77kbq42hjchs72qx7cpp' '-DCMAKE_BUILD_TYPE:STRING=Release' '-DCMAKE_VERBOSE_MAKEFILE:BOOL=ON' '-DCMAKE_FIND_FRAMEWORK:STRING=LAST' '-DCMAKE_FIND_APPBUNDLE:STRING=LAST' '-DCMAKE_INSTALL_RPATH_USE_LINK_PATH:BOOL=FALSE' '-DCMAKE_INSTALL_RPATH:STRING=/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/tasmanian-7.0-u55sduzuuzqw77kbq42hjchs72qx7cpp/lib;/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/tasmanian-7.0-u55sduzuuzqw77kbq42hjchs72qx7cpp/lib64;/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/llvm-openmp-9.0.0-kkckpcgbwofd7nmpry7nq2bdgq3gvdec/lib' '-DCMAKE_PREFIX_PATH:STRING=/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/mpich-3.3.1-dsalg5laudrzwm2svunxhcvq5xsrnigh;/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/openblas-0.3.7-5laehbhhughfsh5rv7iosk56j3lcecof;/Users/balay/spack/opt/spack/darwin-catalina-x86_64/clang-11.0.0-apple/llvm-openmp-9.0.0-kkckpcgbwofd7nmpry7nq2bdgq3gvdec' '-DUSE_XSDK_DEFAULTS:BOOL=ON' '-DXSDK_ENABLE_PYTHON:BOOL=OFF' '-DTPL_ENABLE_MPI:BOOL=ON' '-DXSDK_ENABLE_OPENMP:BOOL=ON' '-DTPL_ENABLE_BLAS:BOOL=ON' '-DXSDK_ENABLE_CUDA:BOOL=OFF' '-DTPL_ENABLE_MAGMA:BOOL=OFF' '-DXSDK_ENABLE_FORTRAN:BOOL=OFF'
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- Check for working CXX compiler: /Users/balay/spack/lib/spack/env/clang/clang++
-- Check for working CXX compiler: /Users/balay/spack/lib/spack/env/clang/clang++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/local/Cellar/cmake/3.15.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.15.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.15.5/share/cmake/Modules/FindOpenMP.cmake:477 (find_package_handle_standard_args)
  Config/CMakeIncludes/sanity_check_and_xsdk.cmake:75 (find_package)
  CMakeLists.txt:111 (include)
```

Reply:
The error is simply that OpenMP option has been turned `ON`, but CMake cannot find OpenMP for this particular compiler.

Apparently the `lvm-openmp` dependencie was not registered.

Reply:
You may also need to add a few additional flags, like `-Xpreprocessor -fopenmp` and `-lomp`. I also needed to add `-I/path/to/llvm-openmp/include` and `-L/path/to/llvm-openmp/lib` for other packages.

Reply:
@adamjstewart CMake is supposed to take care of the OpenMP flags and libs, if CMake doesn't find those, then it would be very hard to pass the flags manually.

Reply:
@mkstoyanov CMake's FindOpenMP.cmake does not seem sufficient to find llvm-openmp when using Apple Clang. See https://github.com/intel/mkl-dnn/issues/591 for more details, and see #13669 for an example of a CMake package where I had to manually override this.

Reply:
@adamjstewart this is good to know for the future, but I think the change is too much for a release that is overdue. It will take quite some time to retest this if I make the changes now.